### PR TITLE
feat: GET Members

### PR DESCRIPTION
**Pull request by simonhlee97**

### CODE REVIEW

File: api/app.ts [hunk 1]
```diff
@@ -17,6 +20,9 @@ app.use(logging({
     overrideSerializers: false
 }));
 app.use(databaseMiddleware());
+app.use(EventRouter.middleware());
+app.use(SeriesRouter.middleware());
+app.use(MembersRouter.middleware());
```
Comment (TheBeege) [inline api/app.ts:25]: These are now covered by `router.middleware()`. These can be removed.

File: api/router.ts [hunk 2]
```diff
@@ -4,28 +4,32 @@ import KoaJoiRouter from 'koa-joi-router';
...
-/*
- * Note that koa-joi-router-docs is using an outdated version of Joi.
- * Upgrading to the latest version (7.0.0) causes a version mismatch error.
- * Please occasionally check if koi-joi-router-docs is updated.
- * Or if you're feeling motivated, submit a PR to update it.
- */
-
 const router = KoaJoiRouter();
 
+router.routes.concat(EventRouter.routes, MembersRouter.routes);
```
Comment (TheBeege) [inline api/router.ts:13]: This can be removed. It's covered by the lines following it.

File: api/members/model.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    id: number;
+
+    @Column({length: 64})
+		firstName: string;
+		
+		@Column({length: 64})
+		lastName: string;
+
+		@Column({length: 140})
+		imageUrl: string;
```
Comment (TheBeege) [inline api/members/model.ts:27]: Why 140?

File: api/members/model.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+		@Column({length: 140})
+		shortBio: string;
```
Comment (TheBeege) [inline api/members/model.ts:30]: Again, why 140 when the database column is text?

File: migrations/1609549880732-CreateMembersTable.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                },
+                {
+                    name: 'updatedAt',
+                    type: 'timestamp',
+                    default: 'CURRENT_TIMESTAMP',
+                    onUpdate: 'CURRENT_TIMESTAMP'
+                }
+            ]
+        }));
+    }
```
Comment (TheBeege) [inline migrations/1609549880732-CreateMembersTable.ts:55]: Okay, this is as great learning opportunity for you 😄 

We have a many-to-many relationship. We have two tables that we want to connect via this relationship - events and members. To create a many-to-many relationship, we need to create a third table. This third table is called a "linking table". You can see an example in TypeORM's explanation in their documentation: https://typeorm.io/#/many-to-many-relations/what-are-many-to-many-relations. In this method, you also don't need to track the events in this table. The linking table tracks them.

However, their default naming convention for the linking table is funky. So back in the MemberModel class, you can update your `@JoinTable()` annotation to include the linking table name: https://github.com/typeorm/typeorm/blob/master/docs/relations.md#jointable-options

To summarize, you'll need to add creation of the linking table in this migration (or create another migration for it. Your choice). You'll also want to remove the `events` field from the `members` table. Following that, you'll want to update the `@JoinTable()` annotation in MemberModel to reference your new linking table's name.

You can also see this article to help understand many-to-many relationships: https://dzone.com/articles/how-to-handle-a-many-to-many-relationship-in-datab

File: api/members/router.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+// TODO: this
+// const Joi = KoaJoiRouter.Joi;
+
+router.prefix('/members');
+
+router.route({
+    path: '/',
+    method: 'GET',
+    handler: Service.listMembers
+});
```
Comment (TheBeege) [inline api/members/router.ts:16]: The Swagger merge added some additional work that needs to be done here. Please check updates to CONTRIBUTING.md and the other routers as a reference. You'll also need to create a `definitions.ts` file.

**Review by TheBeege — CHANGES_REQUESTED** [review]
The formatting of this is all messed up. However, I realize that the linter settings do no specify indentation rules. I'll fix that up after we merge this. In the meantime, can you run VS Code's format on your files? https://code.visualstudio.com/docs/editor/codebasics#_formatting

Looks like Shift+Alt+F will do it.

File: api/members/model.ts (hunk 3 above)
Comment (simonhlee97) [reply api/members/model.ts:27]: paying homage to the original Tweet length

Comment (TheBeege) [reply api/members/model.ts:27]: Ah hahaha okay. Can you update it to 255 to match the database column length?

Can you also updated firstName and lastName to be 80 to match their database columns?

File: api/app.ts (hunk 1 above)
Comment (simonhlee97) [reply api/app.ts:25]: I see.
So I should also remove:
```
import EventRouter from "./event/router";
import SeriesRouter from "./series/router";
import MembersRouter from "./members/router";
```
?

File: api/members/model.ts (hunk 4 above)
Comment (simonhlee97) [reply api/members/model.ts:30]: is it okay to leave blank?
@Column({})

File: api/app.ts (hunk 1 above)
Comment (TheBeege) [reply api/app.ts:25]: Yes, exactly

File: api/members/model.ts (hunk 4 above)
Comment (TheBeege) [reply api/members/model.ts:30]: Yes, you can just to `@column()`. No need to pass in an empty object.

File: migrations/1609549880732-CreateMembersTable.ts (hunk 5 above)
Comment (simonhlee97) [reply migrations/1609549880732-CreateMembersTable.ts:55]: i read the docs and article. I will need some help with this one. Big picture, I understand what many-to-many is, and why another join table is needed. But would like to see a more detailed example than the docs' example.

Comment (kzkaneoka) [reply migrations/1609549880732-CreateMembersTable.ts:55]: @simonhlee97 I hope [this link](https://github.com/typeorm/typeorm/tree/master/sample/sample4-many-to-many/entity) helps😀  It contains many examples about how TypeORM does the many-to-many relationships between the `Post` table and other tables.

> To summarize, you'll need to add creation of the linking table in this migration (or create another migration for it. Your choice). You'll also want to remove the events field from the members table. Following that, you'll want to update the @JoinTable() annotation in MemberModel to reference your new linking table's name.

I want to add one more thing here. I think we don't need a cascade between the `events` and the `members`. @TheBeege please let me know if I'm wrong.
